lib/web_ui/lib/src/engine/window.dart

```diff
@@ -198,6 +203,50 @@ class EngineWindow extends ui.Window {
       callback(data);
     });
   }
+
+
+  /// The setting indicating the current brightness mode of the host platform.
+  /// If the platform has no preference, [platformBrightness] defaults to [Brightness.light].
+  ui.Brightness get platformBrightness => _platformBrightness;
+  ui.Brightness _platformBrightness = ui.Brightness.light;
+
+  /// Reference to css media query that indicates the user theme preference on the web
+  final html.MediaQueryList _brightnessMediaQuery = html.window.matchMedia('(prefers-color-scheme: dark)');
+
+  /// A callback that is invoked whenever [_brightnessMediaQuery] changes value.
+  /// Updates the platformBrightness with the new user preferenceq
+  ///
+  ///  The param brightnessMediaQuery can be a MediaQueryListEvent or a MediaQueryList;
+  ///
+  Function(dynamic) _brightnessMediaQueryListener;
```

Comment:
Using dynamic there doesn’t  convince me at all. 
But when calling the callback for the first time i send it as param _brightnessMediaQuery [[MediaQueryList](https://api.dartlang.org/stable/2.5.0/dart-html/MediaQueryList-class.html)]
Then the listener uses [[MediaQueryListEvent](https://api.dartlang.org/stable/2.5.0/dart-html/MediaQueryListEvent-class.html)] as param. 
Both have the matches param. Maybe there is a better approach

Reply:
True, but i am using the same function with `MediaQueryList` as a param. At the beginning, the listener is not called automatically  so I have to check with the `MediaQueryList` if matches.

Reply:
I think the issue here is that you are trying to call the listener manually for the first time. Instead you could split in 2 methods:
1. `_brightnessMediaQueryListener` which is only called by the browser when media query changes.
2. `_updateMediaBrightness` which you can call initially, and you can also call from inside `_brightnessMediaQueryListener`.

Another thing here is you could use a `typedef` to define the function type.

So basically, something similar to this:
```dart
typedef BrightnessMQListener = void Function(MediaQueryListEvent);

class EngineWindow extends ui.Window {
  BrightnessMQListener _brightnessMediaQueryListener;

  void _addBrightnessMediaQueryListener() {
    _brightnessMediaQueryListener = (MediaQueryListEvent event) {
      _updateBrightness(event.matches ? ui.Brightness.dark : ui.Brightness.light);
    };
    _updateBrightness(_brightnessMediaQuery.matches ? ui.Brightness.dark : ui.Brightness.light);
    _brightnessMediaQuery.addListener(_brightnessMediaQueryListener)
  }
}
```

Reply:
Great, I have done this but now i getting this error:

Uncaught (in promise) Error: Expected a value of type '(Event$) => dynamic', but got one of type '(MediaQueryListEvent$) => Null'

Reply:
Ugh. That's right. The param should be of type `Event`. The workaround would be to take an `html.Event` as a param and then cast it to `html.MediaQueryListEvent`.

You can also remove the `typedef` now.
```dart
class EngineWindow extends ui.Window {
  html.EventListener _brightnessMediaQueryListener;

  void _addBrightnessMediaQueryListener() {
    _brightnessMediaQueryListener = (html.Event event) {
      final MediaQueryListEvent mqEvent = event;
      _updateBrightness(mqEvent.matches ? ui.Brightness.dark : ui.Brightness.light);
    };
    _updateBrightness(_brightnessMediaQuery.matches ? ui.Brightness.dark : ui.Brightness.light);
    _brightnessMediaQuery.addListener(_brightnessMediaQueryListener)
  }
}
```

Reply:
Thanks. It worked!. I have committed the new changes